ovs/lib/migration.py

```diff
@@ -245,6 +245,72 @@ def migrate():
+                                             'write': 'block_cache_on_write',
+                                             'quota': 'quota_bc',
+                                             'backend_prefix': 'backend_bc_{0}'}}
+            if 'arakoon_config' in metadata['backend']:  # Arakoon config should be placed under the backend info
+                metadata['backend']['backend_info']['arakoon_config'] = metadata['backend'].pop('arakoon_config')
+            if 'connection_info' in metadata['backend']:  # Connection info sohuld be placed under the backend info
+                metadata['backend']['backend_info']['connection_info'] = metadata['backend'].pop('connection_info')
+            if 'caching_info' not in metadata:  # Caching info is the new key
+                would_be_caching_info = {}
+                metadata['caching_info'] = would_be_caching_info
+                # Extract all caching data for every storagerouter
+                current_caching_info = metadata['backend'].pop('caching_info')  # Pop to mutate metadata
+                for storagerouter_guid in current_caching_info.iterkeys():
+                    current_cache_data = current_caching_info[storagerouter_guid]
+                    storagerouter_caching_info = {}
+                    would_be_caching_info[storagerouter_guid] = storagerouter_caching_info
+                    for cache_type, cache_type_mapping in structure_map.iteritems():
+                        new_cache_structure = copy.deepcopy(cache_structure)
+                        storagerouter_caching_info[cache_type] = new_cache_structure
+                        for new_structure_key, old_structure_key in cache_type_mapping.iteritems():
+                            if new_structure_key == 'backend_prefix':
+                                # Get possible backend related info
+                                metadata_key = old_structure_key.format(storagerouter_guid)
+                                if metadata_key not in metadata:
+                                    continue
+                                backend_data = metadata.pop(metadata_key)  # Pop to mutate metadata
+                                new_cache_structure['is_backend'] = True
+                                # Copy over the old data
+                                new_cache_structure['backend_info']['arakoon_config'] = backend_data['arakoon_config']
+                                new_cache_structure['backend_info'].update(backend_data['backend_info'])
+                                new_cache_structure['backend_info']['connection_info'].update(backend_data['connection_info'])
+                            else:
+                                new_cache_structure[new_structure_key] = current_cache_data.get(old_structure_key)
+            return metadata
+
+        vpools = VPoolList.get_vpools()
+        for vpool in vpools:
+            new_metadata = _update_metadata_structure(vpool.metadata)
```

Comment:
Can potentially throw KeyErrors and perhaps other errors too, so i'd add this in a try except